Core/Src/main.c

```diff
@@ -87,17 +92,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
+
+  // Starts PWM
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t DataReceived[DATASIZE] = {0x00, 0x80, 0x00};
+  uint8_t DataTransmitted[DATASIZE] = {0x00};
```

Comment:
I think you named them confusingly. The `DataReceived` variable here is actually the data that will be sent out from MCU. The value is also a bit off. 
![image](https://github.com/UWARG/embedded-bootcamp/assets/106219057/a69cfbfa-c52b-4cec-9c03-ad71b7c2d8a9)
From the datasheet, it is saying the first byte you should transmit should be a 0x01 to indicate the transmission start. You have the second byte correct.